dbusutil/dbustest/dbustest.go

```diff
@@ -64,42 +64,42 @@ type DBusHandlerFunc func(msg *dbus.Message, n int) ([]*dbus.Message, error)
 type testDBusStream struct {
 	handler DBusHandlerFunc
 
-	m        sync.Mutex
-	readable sync.Cond
+	outputBuf bytes.Buffer
 
-	outputBuf, inputBuf bytes.Buffer
-	closed              bool
-	authDone            bool
-	n                   int
+	output       chan []byte
+	closeRequest chan struct{}
+
+	closed   atomic.Value
+	authDone bool
+	n        int
 }
 
-func (s *testDBusStream) decodeRequest() {
+func (s *testDBusStream) decodeRequest(req []byte) {
+	buf := bytes.NewBuffer(req)
 	// s.m is locked
```

Comment:
the comments about s.m are no longer needed

Reply:
removed